hilti/toolchain/src/compiler/validator.cc

```diff
@@ -136,6 +136,8 @@ struct VisitorPre : visitor::PreOrder, public validator::VisitorMixIn {
 struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
     using hilti::validator::VisitorMixIn::VisitorMixIn;
 
+    std::unordered_set<uint32_t> _method_declarations; // tracks methods already seen
```

Comment:
Maybe add a comment that this is based on the `ast::DeclarationIndex` if that won't get hashed? This just feels a little unintuitive to me until I looked at the code that used it. That could also just be my lack of experience with that in the Spicy code, though :)

Reply:
Even better: I've made the `DeclarationIndex` hashable, so that we can use it as the index directly.